src/renderer/ipc/index.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { ipcRenderer, IpcRendererEvent } from "electron";
+import { areArgsUpdateAvailableFromMain, UpdateAvailableChannel, onCorrect, UpdateAvailableFromMain, BackchannelArg } from "../../common/ipc";
+import { Notifications, notificationsStore } from "../components/notifications";
+import { Button, ButtonPannel } from "../components/button";
+import { isMac } from "../../common/vars";
+import * as uuid from "uuid";
+
+function UpdateAvailableHandler(event: IpcRendererEvent, ...[backchannel, updateInfo]: UpdateAvailableFromMain): void {
+  const notificationId = uuid.v4();
+
+  function sendToBackchannel(data: BackchannelArg): void {
+    notificationsStore.remove(notificationId);
+    console.log("sending to backchanel", { backchannel, data });
+    ipcRenderer.send(backchannel, data);
+  }
+
+  function renderYesButtons() {
+    if (isMac) {
+      /**
+       * auto-updater's "installOnQuit" is not applicable for macOS as per their docs.
+       *
+       * See: https://github.com/electron-userland/electron-builder/blob/master/packages/electron-updater/src/AppUpdater.ts#L27-L32
+       */
+      return <Button light label="Yes" onClick={() => sendToBackchannel({ doUpdate: true, now: true })} />;
+    }
+
+    return (
+      <>
+        <Button light label="Yes, now" onClick={() => sendToBackchannel({ doUpdate: true, now: true })} />
+        <Button primary outlined label="Yes, later" onClick={() => sendToBackchannel({ doUpdate: true, now: false })} />
+      </>
+    );
+  }
+
+  Notifications.info(
+    (
+      <>
+        <b>Update Available</b>
+        <p>Version {updateInfo.version} of Lens IDE is now available. Would you like to update?</p>
+        <ButtonPannel>
+          {renderYesButtons()}
+          <Button primary outlined label="No" onClick={() => sendToBackchannel({ doUpdate: false })} />
+        </ButtonPannel>
+      </>
+    ), {
+      id: notificationId,
+      onClose() {
+        sendToBackchannel({ doUpdate: false });
+      }
+    }
+  );
+}
+
+export function registerIpcHandlers() {
+  onCorrect(ipcRenderer, UpdateAvailableChannel, UpdateAvailableHandler, areArgsUpdateAvailableFromMain);
```

Comment:
I would more expect to see something:

```ts
return unregisterIpcHandlers(ipcRenderer, UpdateAvailableChannel, UpdateAvailableHandler, areArgsUpdateAvailableFromMain); // returns unsubscribe: () => void
```

Reply:
Why do you expect that the handlers will be removed? These are supposed to be present for the entire lifetime of the `renderer`.

Reply:
Btw, there is not big correlation with `onCorrect` arguments and their names.

_Solution:_ 

Would be better/more readable if `onCorrect` could accept object-map (and this declaration actually easier to extend later when needed, especially critical when you export something like to extensions-api).

```ts
onCorrect({
  source: ipcRenderer,
  channel: UpdateAvailableChannel,
  listener: UpdateAvailableHandler,
  verifier: areArgsUpdateAvailableFromMain
})
```

Also when I open a file with `areArgsUpdateAvailableFromMain` there is also nothing about that it's some kinda `verifier` argument. Maybe better to name file with suffix `*.verifier.ts`? wdyt?

Reply:
Do you mean a verifier function? I figured that it was obvious from the signature (specifically its declared return value). But I can move it to a different function.

As for taking an object instead of positional args. Sure.